### PR TITLE
Jdk8 upgrade

### PR DESCRIPTION
**Pull request by pierre**
This also add Redis dependencies for future use... 😇

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -74,9 +74,6 @@
             <archive>http://groups.google.com/group/killbilling-dev</archive>
         </mailingList>
     </mailingLists>
-    <prerequisites>
-        <maven>3.0</maven>
-    </prerequisites>
```
Comment (sbrossie) [inline pom.xml]: Any reason for dropping the `prerequisites` ?

File: pom.xml [hunk 2]
```diff
@@ -89,11 +86,9 @@
     <properties>
         <async-http-client.version>1.9.40</async-http-client.version>
         <ehcache.version>3.3.1</ehcache.version>
-        <!-- Release 21.0 requires JDK 1.8 or newer -->
-        <guava.version>20.0</guava.version>
+        <guava.version>26.0-jre</guava.version>
```
Comment (sbrossie) [inline pom.xml]: Yeah!

File: pom.xml [hunk 3]
```diff
@@ -103,16 +98,15 @@
         <killbill-client.version>1.0.6</killbill-client.version>
         <!-- This should point to the latest non-SNAPSHOT version of commons -->
         <killbill-commons-for-tests.version>0.22.1</killbill-commons-for-tests.version>
-        <killbill-commons.version>0.22.1</killbill-commons.version>
-        <killbill-platform.version>0.38.2</killbill-platform.version>
+        <killbill-commons.version>0.22.2-SNAPSHOT</killbill-commons.version>
+        <killbill-platform.version>0.38.3-SNAPSHOT</killbill-platform.version>
```
Comment (sbrossie) [inline pom.xml]: Do we need a snapshot ?

File: pom.xml [hunk 4]
```diff
@@ -103,16 +98,15 @@
...
         <killbill-plugin-api.version>0.25.0</killbill-plugin-api.version>
         <logback.version>1.2.3</logback.version>
         <metrics-palominolabs.version>3.2.1</metrics-palominolabs.version>
         <metrics.version>3.2.2</metrics.version>
         <project.build.sourceEncoding>UTF-8</project.build.sourceEncoding>
         <slf4j.version>1.7.25</slf4j.version>
-        <!-- Release 1.5.11+ requires Jackson 2.8+ -->
-        <swagger.version>1.5.10</swagger.version>
+        <swagger.version>1.5.21</swagger.version>
```
Comment (sbrossie) [inline pom.xml]: Nice.

File: pom.xml [hunk 5]
```diff
@@ -744,7 +748,7 @@
             <dependency>
                 <groupId>org.javassist</groupId>
                 <artifactId>javassist</artifactId>
-                <version>3.19.0-GA</version>
+                <version>3.21.0-GA</version>
```
Comment (sbrossie) [inline pom.xml]: Unrelated to your PR, but do you remember what we are using `javassist` for ?

File: pom.xml [hunk 6]
```diff
@@ -2219,9 +2231,6 @@
         </profile>
         <profile>
             <id>jdk16</id>
```
Comment (sbrossie) [inline pom.xml]: Do we still need this profile `jdk16 ` ?

File: pom.xml (hunk 1 above)
Comment (pierre) [reply pom.xml]: `prerequisites` as a top-level section is only meaningful for Maven plugins. Instead, we should be using the `maven-enforcer-plugin` if we really wanted to enforce the Maven version.

I looked into it but:

* I don't fully remember why we needed Maven 3.0+ (could be related to some of the plugins)
* Maven 3.0 was released back in 2010, so we should be safe :-)

File: pom.xml (hunk 3 above)
Comment (pierre) [reply pom.xml]: This is only to make CircleCI happy. If a project (e.g. `killbill`) has a snapshot dependency on branch `feature-X`, we will pull and build locally dependencies (e.g. `killbill-platform` and `killbill-commons`) on branch `feature-X` (if it exists). I have a few branches `jdk8-upgrade` in other repos coming up 😄 

This is why https://github.com/killbill/killbill/pull/1052 is green for instance (there is a `killbill-oss-parent` `caching-improvements` branch).

Will cleanup before merging and releasing.

File: pom.xml (hunk 5 above)
Comment (pierre) [reply pom.xml]: It's a transitive dependency (e.g. by Swagger). We declare it to fix some versioning conflicts.

File: pom.xml (hunk 6 above)
Comment (pierre) [reply pom.xml]: Nope (we could remove `jdk17` too) but that would require some further cleanup (CircleCI, etc.). Would rather punt it for now but I can come back to it.

**Pull request merged**